src/main/java/ponyo/storage/Storage.java

```diff
@@ -0,0 +1,178 @@
+package ponyo.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+import ponyo.data.exceptions.PonyoException;
+import ponyo.data.task.Deadline;
+import ponyo.data.task.Event;
+import ponyo.data.task.Task;
+import ponyo.data.task.TaskList;
+import ponyo.data.task.Todo;
+
+
+/**
+ * Handles loading and saving of tasks into the file created.
+ */
+public class Storage {
+    private static final String PATH = "src/main/data";
+    private static final String FILENAME = "tasks.txt";
+
+    protected String filePath;
+
+    /**
+     * @param filePath file path of the stored file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * @return the list of tasks stored in hard disk.
+     */
+    public ArrayList<Task> load() {
+        try {
+            ArrayList<Task> tasks = new ArrayList();
+            readTasksFromFile(tasks, filePath);
+            return tasks;
+        } catch (IOException e) {
+            throw new PonyoException("Error loading from file :(");
+        }
+    }
+
+    /**
+     * Checks if the file and folder exists
+     *
+     * @param filePath path to check
+     * @throws IOException if file creation fails
+     */
+    public void checkIfFileFolderExists(String filePath) throws IOException {
+        File file = new File(filePath);
+        File directory = new File(PATH);
+
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+    }
+
+    /**
+     * Read the tasks from the file separated with delimiter (-)
+     *
+     * @param tasks stores the list of tasks read from the file
+     * @param filePath path of file
+     * @throws IOException if error reading file
+     */
+    public void readTasksFromFile(ArrayList tasks, String filePath) throws IOException {
```

Comment:
Also not naming related but I am curious why you decided to take in the `ArrayList tasks` as a parameter instead of either returning a newly created `ArrayList` or returning the raw text from `Stoage` class and parsing it in your `Tasks` or `TaskList` class.

Reply:
Ah you're right! Made a rookie mistake of mutating my arguments in the process. 👎  Thanks! :)